sunpy/map/tests/test_mapbase.py

```diff
@@ -416,9 +429,39 @@ def test_shift_history(generic_map):
     assert resultant_shift[1] == y_shift1 + y_shift2
 
 
+@pytest.mark.parametrize('rect, submap_out',
```

Comment:
I think we might want to expand this test to a 3x3 array. I think this is what the discussion above was getting at, but `np.round(0.5)=0` while `np.round(1.5)=2`, meaning `[(0,0),(1.5,1.5)]` would actually return the whole 3-by-3 array when it should return only `[0:2,0:2]`. So as it stands, this test would fail if extended beyond a 2x2. We should make sure we're testing a general enough case.

Reply:
I've updated the rounding to get rid of `np.round` completely - I can still add a 3x3 test map?

Reply:
Yeah, I think it would still be a good idea to have a pixel center with a coordinate >1. Presumably this should just be changing the `simple_map` fixture to `np.arange(9).reshape(3,3)` and then adding one line to the test parametrization here?